src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,480 @@
+- specifying default values for optional fields using `.default()`
+- checking the *shape* of string values with built-in regexes, like `.url()` for URLs and `.email()` for emails
+- transforming a frontmatter value into another value, like parsing a publish date to [a `Date` object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date).
+
+```ts
+{
+  // "language" is optional, with a default value of english
+  language: z.enum(['es', 'de', 'en']).default('en'),
+  // allow email strings only.
+  // ex. "jeff" would fail to parse, but "hey@blog.biz" would pass
+  authorContact: z.string().email(),
+  // allow url strings only.
+  // ex. "/post" would fail, but `https://blog.biz/post` would pass
+  canonicalURL: z.string().url(),
+  // parse publishDate as a browser-standard Date object
+  // ex. "Bananaday" would fail to parse, but "2022-10-10" would pass
+  publishDate: z.string().transform(str => new Date(str)),
+}
+```
+
+You can [browse Zod's documentation](https://github.com/colinhacks/zod) for a complete rundown of features.
+
+### Zod quick reference
+
+Frontmatter YAML covers strings, booleans, numbers, objects, and arrays. Here's a quick cheatsheet for each of those:
+
+```ts
+import { z, defineCollection } from 'astro:content';
+
+defineCollection({
+  schema: {
+    // Boolean
+    isDraft: z.boolean(),
+    // String
+    title: z.string(),
+    // Number
+    sortOrder: z.number(),
+    // Optional - Extend any type with `.optional()`
+    footnote: z.string().optional(),
+    // Default value - Extend any type with `.default(value)`
+    author: z.string().default('Anonymous'),
+    // Array
+    // Note: array(...) is a wrapper around the type of each element
+    // Ex. array of strings:
+    tags: z.array(z.string()),
+    // Enum
+    language: z.enum(['en', 'es']),
+    // Object
+    image: z.object({
+      src: z.string(),
+      alt: z.string(),
+    }),
+  }
+})
+```
+
+See the [**Why Zod?** example](#why-zod) above for more Zod-specific features like transforms.
+
+## Getting content
+
+Astro provides 2 functions to query collections:
+
+- `getCollection` - get all entries in a collection, or based on a filter
+- `getEntry` - get a specific entry in a collection by file name
+
+These functions will have typed based on collections that exist. In other words, `getCollection('banana')` will raise a type error if there is no `src/content/banana/`.
+
+```astro
+---
+import { getCollection, getEntry } from 'astro:content';
+// Get all `blog` entries
+const allBlogPosts = await getCollection('blog');
+// Filter blog posts by entry properties
+const draftBlogPosts = await getCollection('blog', ({ id, slug, data }) => {
+  return data.status === 'draft';
+});
+// Get a specific blog post by file name
+const enterprise = await getEntry('blog', 'enterprise.md');
+---
+```
+
+### Return type
```

Comment:
Note: will likely move this to API reference when that PR is ready

Reply:
Update: you know what? I agree this is pretty useful. I simplified the schema a bit and tweaked some wording, and I'm happy to keep as-is. Flagging for review @delucis